bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/handler/GardenaAccountHandler.java

```diff
@@ -51,80 +62,191 @@
 
+    // assets
+    private @Nullable GardenaDeviceDiscoveryService discoveryService;
     private @Nullable GardenaSmart gardenaSmart;
-    private HttpClientFactory httpClientFactory;
-    private WebSocketFactory webSocketFactory;
+    private final HttpClientFactory httpClientFactory;
+    private final WebSocketFactory webSocketFactory;
+    private final Bundle bundle;
+    private final TranslationProvider i18nProvider;
+    private final LocaleProvider localeProvider;
 
-    public GardenaAccountHandler(Bridge bridge, HttpClientFactory httpClientFactory,
-            WebSocketFactory webSocketFactory) {
+    // re- initialisation stuff
+    private final Object reInitializationCodeLock = new Object();
+    private @Nullable ScheduledFuture<?> reInitializationTask;
+    private boolean reInitializationCausedBy429 = false;
+    private Instant lastApiCallTime = Instant.MIN;
+    private boolean lastApiCallTimeLoaded = false;
+
+    public GardenaAccountHandler(Bridge bridge, HttpClientFactory httpClientFactory, WebSocketFactory webSocketFactory,
+            Bundle bundle, TranslationProvider i18nProvider, LocaleProvider localeProvider) {
         super(bridge);
         this.httpClientFactory = httpClientFactory;
         this.webSocketFactory = webSocketFactory;
+        this.bundle = bundle;
+        this.i18nProvider = i18nProvider;
+        this.localeProvider = localeProvider;
+    }
+
+    /**
+     * Returns the time when the last api call was made. If the thing was newly created, restores the time from a
+     * property to ensure consistent behaviour over restarts.
+     *
+     * @return the time when the last api call was made.
+     */
+    private Instant lastApiCallTime() {
+        if (!lastApiCallTimeLoaded) {
+            Map<String, String> properties = getThing().getProperties();
+            String property = properties.getOrDefault(GardenaBindingConstants.LAST_API_CALL_TIME, "");
+            lastApiCallTime = "".equals(property) ? Instant.now().minus(Duration.ofHours(1)) : Instant.parse(property);
+            lastApiCallTimeLoaded = true;
+        }
+        return lastApiCallTime;
+    }
+
+    /**
+     * Updates the time when the last api call was made to now(). Saves the value to a property to ensure consistent
+     * behaviour over restarts.
+     */
+    private void lastApiCallTimeUpdate() {
+        lastApiCallTime = Instant.now();
+        getThing().setProperty(GardenaBindingConstants.LAST_API_CALL_TIME,
+                lastApiCallTime.truncatedTo(ChronoUnit.SECONDS).toString());
     }
 
     @Override
     public void initialize() {
         logger.debug("Initializing Gardena account '{}'", getThing().getUID().getId());
-        initializeGardena();
+        Instant now = Instant.now();
+        Instant notBeforeTime = lastApiCallTime().plus(REINITIALIZE_DELAY_MINUTES_BACK_OFF)
+                .plus(REINITIALIZE_DELAY_MINUTES_BACK_OFF);
+        if (now.isBefore(notBeforeTime)) {
```

Comment:
Readability: "is before not before time" is somehow hard for me to read. :) Consider inverting the logic, for example by renaming variable to `scheduledAfter`, then:
```suggestion
        if (now.isAfter(scheduledAfter)) {
```
(and swapping the two blocks, of course)

Reply:
Api call suppression concept modified.